### PR TITLE
Drop Machine Controller Manager management

### PR DESCRIPTION
**Pull request by AndreasBurger**
**How to categorize this PR?**
/area open-source
/kind cleanup
/platform openstack

**What this PR does / why we need it**:
With the release of Gardener `v1.83.0` the MCM deployment is handled by gardener, so the corresponding code in the extensions can safely be dropped.

**Special notes for your reviewer**:
Some related code was already dropped with recent gardener-upgrade-PRs

**Release note**:

```other developer
The code related to `machine-controller-manager` management has been cleaned up because `gardenlet` is responsible for it since `gardener/gardener@v1.83`.
```

### CODE REVIEW

File: pkg/webhook/controlplane/ensurer.go [hunk 1]
```diff
@@ -59,43 +56,6 @@ type ensurer struct {
...
-	if newObj.Spec.ResourcePolicy == nil {
-		newObj.Spec.ResourcePolicy = &vpaautoscalingv1.PodResourcePolicy{}
-	}
-
-	newObj.Spec.ResourcePolicy.ContainerPolicies = extensionswebhook.EnsureVPAContainerResourcePolicyWithName(
-		newObj.Spec.ResourcePolicy.ContainerPolicies,
-		machinecontrollermanager.ProviderSidecarVPAContainerPolicy(openstack.Name, minAllowed, maxAllowed),
-	)
-	return nil
-}
```
Comment (rfranzke) [inline pkg/webhook/controlplane/ensurer.go:97]: Why is this removed? We need it. PTAL at the Alicloud PR.

**Review by kon-angelo — APPROVED** [review]
/lgtm
